### PR TITLE
Updates REST API for a new data structure capturing Java versions

### LINKED ISSUE #286: Possibility to add Java 11 firstJava11SupportedVersion to REST API
I'm trying to fix [JDK 11 Is Not Allowed In the Project Wizard](https://github.com/MicroShed/mp-starter-intellij-ext/issues/15). So in the new project wizard in IntelliJ, we need to know which implementation supports Java 11 and from which version. Something that we already have in starter UI (the firstJava11SupportedVersion property). Is it possible to expose this property as part of the REST API or should we duplicate such logics in each IDE plugin (vscode, intellij and later netbeans and eclipse). 

It may seems not related to the API at all as it's only used for updating the UI; but actually at the moment the client code in IDE plugin has no sense about any implementation and everything is just created dynamically based on the JSON output of the REST API. That's why we don't need to update the plugin at all when a new implementation or a new version gets introduced.

### PR DESCRIPTION
**Pull request by Karm**
* update to the checkstyle to allow public members without getters and setters
* fixes #286 #264 by adding Java SE version to the description of runtimes
* maintains legacy API v1, 2 and 3

### CODE REVIEW

File: src/main/java/org/eclipse/microprofile/starter/rest/model/ServerOptions.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import org.eclipse.microprofile.starter.core.model.MicroProfileVersion;
+
+import java.util.List;
+
+/**
+ * @author Michal Karm Babacek <karm@redhat.com>
+ */
+public class ServerOptions {
+
+    public final MicroProfileVersion mpVersion;
```
Comment (rdebusscher) [inline src/main/java/org/eclipse/microprofile/starter/rest/model/ServerOptions.java:33]: You can also use private and a getter. Why the change of coding style? (which required a change to checkstyle)

File: src/main/java/org/eclipse/microprofile/starter/core/model/JavaSEVersion.java [hunk 2]
```diff
@@ -26,7 +26,9 @@
  *
  */
 public enum JavaSEVersion implements ComboBoxItem {
-    NONE(null, ""), SE8("1.8", "Java 8"), SE11("11", "Java 11");
+    NONE(null, ""),
```
Comment (rdebusscher) [inline src/main/java/org/eclipse/microprofile/starter/core/model/JavaSEVersion.java:29]: Use 

    // @formatter:off

(and :on) to keep the manual formatting.

File: src/main/java/org/eclipse/microprofile/starter/rest/model/ServerOptions.java (hunk 1 above)
Comment (Karm) [reply src/main/java/org/eclipse/microprofile/starter/rest/model/ServerOptions.java:33]: @rdebusscher I wanted to avoid a getter and setter that does nothing and is not necessary for any proxies etc. I can change it, if you would prefer it.

**Comment by Karm** [comment]
@rdebusscher  nudge nudge

File: src/main/java/org/eclipse/microprofile/starter/rest/model/ServerOptions.java (hunk 1 above)
Comment (rdebusscher) [reply src/main/java/org/eclipse/microprofile/starter/rest/model/ServerOptions.java:33]: But there are already similar parts where getters and setters are created.

So this will introduce several coding styles in one project which is never beneficial for readability. (because people will start to wonder why we did this)
